test/agentchat/test_groupchat.py

```diff
@@ -1176,6 +1176,121 @@ def custom_speaker_selection_func(last_speaker: Agent, groupchat: GroupChat) ->
     assert "teamA_executor" in speakers
 
 
+def test_select_speaker_message_and_prompt_templates():
+    """
+    In this test, two agents are part of a group chat which has customized select speaker message and select speaker prompt templates. Both valid and empty string values will be used.
+    The expected behaviour is that the customized speaker selection message and prompts will override the default values or throw exceptions if empty.
+    """
+
+    agent1 = autogen.ConversableAgent(
+        "Alice",
+        description="A wonderful employee named Alice.",
+        human_input_mode="NEVER",
+        llm_config=False,
+    )
+    agent2 = autogen.ConversableAgent(
+        "Bob",
+        description="An amazing employee named Bob.",
+        human_input_mode="NEVER",
+        llm_config=False,
+    )
+
+    # Customised message, this is always the first message in the context
+    custom_msg = """You are the CEO of a niche organisation creating small software tools for the healthcare sector with a small team of specialists. Call them in sequence.
+    The job roles and responsibilities are:
+    {roles}
+
+    You must select only from {agentlist}."""
+
+    # Customised prompt, this is always the last message in the context
+    custom_prompt = """Read the above conversation.
+    Then select the next job role from {agentlist} to take action.
+    RETURN ONLY THE NAME OF THE NEXT ROLE."""
+
+    # Test empty is_termination_msg function
+    groupchat = autogen.GroupChat(
+        agents=[agent1, agent2],
+        messages=[],
+        speaker_selection_method="auto",
+        max_round=10,
+        select_speaker_message_template=custom_msg,
+        select_speaker_prompt_template=custom_prompt,
+    )
+
+    # Test with valid strings, checking for the correct string and roles / agentlist to be included
+
+    assert (
+        groupchat.select_speaker_msg()
+        == """You are the CEO of a niche organisation creating small software tools for the healthcare sector with a small team of specialists. Call them in sequence.
+The job roles and responsibilities are:
+Alice: A wonderful employee named Alice.
+Bob: An amazing employee named Bob.
+
+You must select only from ['Alice', 'Bob']."""
+    )
+
+    assert (
+        groupchat.select_speaker_prompt()
+        == """Read the above conversation.
+Then select the next job role from ['Alice', 'Bob'] to take action.
+RETURN ONLY THE NAME OF THE NEXT ROLE."""
+    )
+
+    # Test with empty strings
+    try:
```

Comment:
Instead of using try-except, you can use pytest.

```python
import pytest

def function_that_raises_value_error():
    raise ValueError("An example error")

def test_function_that_raises_value_error():
    with pytest.raises(ValueError) as e:
        function_that_raises_value_error()
    assert "An example error" in str(e.value)
```

Reply:
Okay, I've changed the string comparison and changed the exception check using pytest